src/Illuminate/Http/Resources/ConditionallyLoadsAttributes.php

```diff
@@ -28,15 +26,30 @@ protected function filter($data)
             }
 
             if (is_numeric($key) && $value instanceof MergeValue) {
-                return $this->mergeData($data, $index, $this->filter($value->data), $numericKeys);
+                return $this->mergeData($data, $index, $this->filter($value->data));
```

Comment:
This behaves differently than #27325:
#27325 determines `$numericKeys` on `$value->data`, your PR on `$this->filter($value->data)`.

You can see the incorrect result by adjusting the data in `test_leading_merge__keyed_value_is_merged_correctly`:

```php
new MergeValue([0 => new MissingValue, 'name' => 'mohamed', 'location' => 'hurghada'])
```

Reply:
Thanks, I'll take a look tomorrow.